Lib/traceback.py

```diff
@@ -1532,7 +1532,11 @@ def _compute_suggestion_error(exc_value, tb, wrong_name):
         # has the wrong name as attribute
         if 'self' in frame.f_locals:
             self = frame.f_locals['self']
-            if hasattr(self, wrong_name):
+            try:
+                has_wrong_name = hasattr(self, wrong_name)
+            except Exception:
```

Comment:
Why do we need to handle exceptions here? hasattr returns a boolean.

Reply:
Because `hasattr` triggers `__getattr__` which can raise an exception. If it raises - the whole repl process will fail with an error.